### PR TITLE
Emsg payload

### PR DESCRIPTION
**Pull request by anbangisak**
Emsg box written to read contents from eMsg box of mp4 file.

### CODE REVIEW

File: EmsgBox.go [hunk 1]
```diff
@@ -0,0 +1,126 @@
+package mp4box
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+)
+
+//FullBox - Base box holding the bytes
```
Comment (eswarantg) [inline EmsgBox.go]: What is the structure of the box...
Please capture the payload structure

Comment (eswarantg) [reply EmsgBox.go]: https://aomediacodec.github.io/id3-emsg/

File: EmsgBox.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+/*
+aligned(8) class FullBox(unsigned int(32) boxtype, unsigned int(8) v, bit(24) f) extends Box(boxtype) {
+	unsigned int(8) version = v;
+	bit(24) flags = f;
+}
+*/
+type EmsgBox struct {
+	BaseBox
```
Comment (eswarantg) [inline EmsgBox.go]: From FullBox not BaseBox

File: EmsgBox.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+//GetTimeScale - Returns the timescale data from payload
+func (b *EmsgBox) GetTimeScale() []byte {
+	return b.TimeScale
+}
+
+func (b *EmsgBox) GetAllData() (schemeIdUri string, schemeIdVal string, msgData string,
+	timeScale []byte, presentTimeDelta []byte, evtDur []byte, idVal []byte) {
+	payload := b.BaseBox.getPayload()
```
Comment (eswarantg) [inline EmsgBox.go]: you should get FullBox Payload that strips the header and gives you the base slice with only this payload

File: EmsgBox.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+		startPayload := payload[20:]
+		schemeIdUriIdx := bytes.IndexByte(startPayload, 0)
+		schemeIdUri = string(startPayload[0:schemeIdUriIdx])
+
+		schemeIdValPayload := startPayload[schemeIdUriIdx+1:]
+		schemeIdValIdx := bytes.IndexByte(schemeIdValPayload, 0)
+		schemeIdVal = string(schemeIdValPayload[0:schemeIdValIdx])
+
+	}
+	return
```
Comment (eswarantg) [inline EmsgBox.go]: Is there an error condition here if vesion is neither 1 or 0

File: EmsgBox.go [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+}
+
+//Version - returns the version of the box
+func (b *EmsgBox) Version() int8 {
```
Comment (eswarantg) [inline EmsgBox.go]: You shall not need this function as the FullBox will already have Version

File: EmsgBox.go [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+	if len(p) >= 1 {
+		return int8(p[0])
+	}
+	//Improper Box
+	return ret
+}
+
+//Flags - Returns flags
+// bit(24) - fit to lower 24 bits of uint32
+func (b *EmsgBox) Flags() uint32 {
```
Comment (eswarantg) [inline EmsgBox.go]: Flags again is available in fullBox

File: EmsgBox.go (hunk 1 above)
Comment (anbangisak) [reply EmsgBox.go]: updated the payload structure to the code.

File: EmsgBox.go (hunk 2 above)
Comment (anbangisak) [reply EmsgBox.go]: changed Basebox to Fullbox

File: EmsgBox.go (hunk 3 above)
Comment (anbangisak) [reply EmsgBox.go]: updated basebox payload to fullbox payload.

File: EmsgBox.go (hunk 4 above)
Comment (anbangisak) [reply EmsgBox.go]: added error condition, if version is neither 0 or 1.

File: EmsgBox.go (hunk 5 above)
Comment (anbangisak) [reply EmsgBox.go]: Removed Version function since FullBox already has it.

File: EmsgBox.go (hunk 6 above)
Comment (anbangisak) [reply EmsgBox.go]: Removed Flags function since FullBox already has it.

**Review by eswarantg — APPROVED** [review]

**Pull request merged**